packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -0,0 +1,405 @@
+
+1. `editor` section is the actual text area to add or update rich text content.
+2. `footer` section consists of `nimble-toggle-button` to control each text formatting functionalities like bold, italic, etc, and a
+   `footer-actions` slot element which is typically used to add action buttons to the right bottom of the component.
+
+Example usage of the `nimble-rich-text-editor` in the application layer is as follows:
+
+```html
+<nimble-rich-text-editor>
+    <nimble-button slot="footer-actions">Cancel</nimble-button>
+    <nimble-button slot="footer-actions">Add Comment</nimble-button>
+</nimble-rich-text-editor>
+```
+
+### API
+
+_Props/Attrs_
+
+-   `markdownValue` - is an accessor used to get and set the markdown value.
+    -   `getter` - this will serialize the content by extracting the Node from the editor and convert it into a markdown string using
+        [prosemirror-markdown serializer](https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/to_markdown.ts#L30).
+    -   `setter` - this will parse the markdown string into a Node and load it back into the editor using
+        [prosemirror-markdown parser](https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/from_markdown.ts#L199).
+-   `isEmpty` - is a read-only property that indicates whether the editor is empty or not. This will be achieved through Tiptap's
+    [isEmpty](https://tiptap.dev/api/editor#is-empty) API.
+
+_Methods_
+
+-   none
+
+_Events_
+
+-   `input` - event emitted when there is a change in the editor. This can be achieved through Tiptap's [update event](https://tiptap.dev/api/events#update).
+    Below are a few scenarios to understand when this event will fire:
+
+    1. This event will fired for every input in the content of the editor, including text inputs, text formatting changes, and text removals.
+    2. This event will not fire when there are no changes made to the content of the editor. For example, all mouse events, selecting the texts, state
+       changes, etc,
+
+_CSS Classes and CSS Custom Properties that affect the component_
+
+-   The minimum height of the component will be set to show at least one line in the text area along with a footer section. If the content exceeds
+    the current height (set by the client), a vertical scrollbar will appear, allowing users to view the hidden content.
+-   The minimum width of the component will be determined based on ensuring all buttons in the footer are visible. We have arrived to this decision
+    to align with the minimal requirement for the initial release. However, adjustments will be made later based on the visual design for mobile view is complete.
+-   The content in the text area will adjust its layout based on the width, potentially increasing the height of the editor. But in case of having
+    a long nested lists that is beyond the current width (set by the client), a horizontal scrollbar will appear within the text area to view the content.
```

Comment:
Is this saying the control will always grow vertically to fit the content that is being edited without making a vertical scrollbar?

I think we should be using the native textarea as an example of the default behavior we should follow. If I understand correctly that the intention is the control will grow to fit the content I think that should be an optional configuration. Maybe a `<nimble-rich-text-editor auto-size>` or a `<nimble-rich-text-editor fit-to-content>` option

This applies to the viewer element as-well